src/solver/conv_MP_bidirectional_winograd.cpp

```diff
@@ -697,17 +752,20 @@ conv::DataInvokeParams GetTransformedInvokeContext(const ConvolutionContext& ctx
                                                    const AnyInvokeParams& invoke_ctx)
 {
 #if MIOPEN_BACKEND_HIP
+    const miopenDataType_t transform_data_type =
+        miopen::IsEnabled(MIOPEN_DEBUG_AMD_MP_BD_WINOGRAD_ENABLE_FP16_TRANSFORM{})
+            ? ctx.in_data_type
+            : miopenFloat;
     WinogradBufferInfo<WinoDataH, WinoFilterH, WinoDataW, WinoFilterW>
         wino_in = GetWinoBuffer<WinoDataH, WinoFilterH, WinoDataW, WinoFilterW>(
-            ctx, ConvWinoBuffType::Input),
+            ctx, ConvWinoBuffType::Input, transform_data_type),
         wino_out = GetWinoBuffer<WinoDataH, WinoFilterH, WinoDataW, WinoFilterW>(
-            ctx, ConvWinoBuffType::Output),
+            ctx, ConvWinoBuffType::Output, transform_data_type),
         wino_wei = GetWinoBuffer<WinoDataH, WinoFilterH, WinoDataW, WinoFilterW>(
-            ctx, ConvWinoBuffType::Weight);
-
+            ctx, ConvWinoBuffType::Weight, transform_data_type);
+    // tidy error
```

Comment:
This is misleading comment, because the suppression below is for cppcheck.

Reply:
This comment mean:there are no unread variables, but compiler still shows a warning, so let's suppress them.

Reply:
If you would like to say that this suppression is due to a bug in `cppcheck` just say so.

What I see now:
- The 1st comment says that `clang-tidy` issues an error
- The 2nd comment suppresses `cppcheck` error

This confuses me.

Reply:
Resolved.